### PR TITLE
Add EffForceAttack

### PR DESCRIPTION
**Pull request by OfficialDonut**
### Description
Add force attack effect to make a living entity attack an entity.

---
**Target Minecraft Versions:** 1.15.2+
**Requirements:** 1.15.2+
**Related Issues:** no

### CODE REVIEW

File: src/main/java/ch/njol/skript/effects/EffForceAttack.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+
+@Name("Force Attack")
+@Description("Makes a living entity attack an entity with a melee attack")
```
Comment (ShaneBeee) [inline src/main/java/ch/njol/skript/effects/EffForceAttack.java:38]: Super small nitpick, but a description should end in a period, to keep consistent with other descriptions

File: src/main/java/ch/njol/skript/effects/EffForceAttack.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+@Examples("spawn a wolf at player's location\nmake last spawned wolf attack player")
```
Comment (ShaneBeee) [inline src/main/java/ch/njol/skript/effects/EffForceAttack.java:39]: In my opinion, for readability this should be split, ex:
```java
@Examples({"spawn a wolf at player's location",
    "make last spawned wolf attack player"})
```

**Review by ShaneBeee — CHANGES_REQUESTED** [review]
Looks good to me. One small request, and one small suggestion for readability.

**Review by ShaneBeee — DISMISSED** [review]

File: src/main/java/ch/njol/skript/effects/EffForceAttack.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	
+	@SuppressWarnings("null")
+	private Expression<LivingEntity> entities;
+	@SuppressWarnings("null")
+	private Expression<Entity> target;
+	
+	@SuppressWarnings("unchecked")
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!ATTACK_IS_SUPPORTED) {
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/effects/EffForceAttack.java:59]: Should be checked in registration.

File: src/main/java/ch/njol/skript/effects/EffForceAttack.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+@Name("Force Attack")
+@Description("Makes a living entity attack an entity with a melee attack.")
+@Examples({"spawn a wolf at player's location",
+	"make last spawned wolf attack player"})
+@Since("INSERT VERSION")
+@RequiredPlugins("1.15.2 or newer")
+public class EffForceAttack extends Effect {
+	
+	static {
+		Skript.registerEffect(EffForceAttack.class, "make %livingentitys% attack %entity%", "force %livingentitys% to attack %entity%");
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/effects/EffForceAttack.java:46]: Patterns should be on their own lines. Also, shouldn't it be `livingentities`? I don't know if that actually matters though.

File: src/main/java/ch/njol/skript/effects/EffForceAttack.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+			return false;
+		}
+		entities = (Expression<LivingEntity>) exprs[0];
+		target = (Expression<Entity>) exprs[1];
+		return true;
+	}
+	
+	@Override
+	protected void execute(Event e) {
+		@Nullable Entity target = this.target.getSingle(e);
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/effects/EffForceAttack.java:70]: We usually don't include annotations like this. It should be okay for the warning to be there.

**Review by APickledWalrus — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/effects/EffForceAttack.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+
+@Name("Force Attack")
+@Description("Makes a living entity attack an entity with a melee attack.")
+@Examples({"spawn a wolf at player's location",
+	"make last spawned wolf attack player"})
+@Since("INSERT VERSION")
+@RequiredPlugins("1.15.2 or newer")
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/effects/EffForceAttack.java:42]:
```suggestion
@RequiredPlugins("Minecraft 1.15.2+")
```

File: src/main/java/ch/njol/skript/effects/EffForceAttack.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+	@SuppressWarnings("null")
+	private Expression<LivingEntity> entities;
+	@SuppressWarnings("null")
+	private Expression<Entity> target;
+	
+	@SuppressWarnings("unchecked")
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!ATTACK_IS_SUPPORTED) {
+			Skript.error("The force attack effect requires server version 1.15.2 or newer");
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/effects/EffForceAttack.java:62]:
```suggestion
			Skript.error("The force attack effect requires Minecraft 1.15.2 or newer");
```

**Review by APickledWalrus — CHANGES_REQUESTED** [review]

**Review by ShaneBeee — APPROVED** [review]

**Review by APickledWalrus — APPROVED** [review]

**Pull request merged**